shared/js/media/media_frame.js

```diff
@@ -51,8 +51,26 @@ function MediaFrame(container, includeVideo, maxImageSize) {
     this.video = new VideoPlayer(container);
     this.video.hide();
   }
+
+  // Add a class to the container so we could find it later and use it as
+  // a key in the instance weakmap.
+  this.container.classList.add('media-frame');
+  MediaFrame.instancesToLocalize.set(container, this);
 }
 
+// WeakMap with the container nodes as keys and MediaFrame instances as values.
+MediaFrame.instancesToLocalize = new WeakMap();
+
+navigator.mozL10n.ready(function() {
+  // Retrieve MediaFrame instances by searching for container nodes.
+  for (var container of document.querySelectorAll('.media-frame')) {
```

Comment:
This is the first time I've see for/of in Gaia.  I was confused at first, but I guess that is the price of progress :-)